doc/guides/investigating_native_memory_leak.md

```diff
@@ -0,0 +1,441 @@
+# Investigating Memory Leaks with valgrind
+
+A Node.js process may run out of memory due to excessive consumption of
+`Native memory`. `Native Memory` is memory which is not managed by the the
+V8 Garbage collector and is allocated either by the Node.js runtime, it's
+dependancies or native [addons](://nodejs.org/docs/latest/api/n-api.html).
+
+This guide provides informaiton on how to use valgrind to investigate these
+issues.
+
+## valgrind
+
+[Valgrind](https://valgrind.org/docs/manual/quick-start.html) is a
+tool available on linux distributions which can be used to investigate
+memory usage including identifying memory leaks (memory which is
+allocated and not freed) and other memory related problems
+like double freeing memory.
+
+To use valgrind:
+
+* Be patient, running under valgrind slows execution significantly due
+  to the checks being performed.
+* Reduce your test case to the smallest reproduce. Due to the slowdown it is
+  important to run the minimum testcase in order to be able to do it in
+  a reasonable time.
+
+## Installation
+
+It is an optional package in most cases and must be installed. For example on
+ubuntu:
+
+```bash
```

Comment:
```suggestion
```console
```

...or `text` or nothing at all, but probably not `bash`. Using ```bash will result in bash syntax highlighting as if the commands are in a bash script.